components/brave_wallet/common/buildflags/buildflags.gni

```diff
@@ -0,0 +1,10 @@
+# Copyright (c) 2025 The Brave Authors. All rights reserved.
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this file,
+# You can obtain one at https://mozilla.org/MPL/2.0/.
+
+import("//brave/components/brave_origin/buildflags/buildflags.gni")
+
+declare_args() {
+  enable_brave_wallet = !is_brave_origin_branded
+}
```

Comment:
This looks like a MEGA-sized PR which we must avoid :) 
Nature of changes is not mechanical at all.

I think we need this `enable_brave_wallet` flag defined and set to `true` for start and iteratively make components/folder depend on it. Then finalize by tying to `is_brave_origin_branded`.

Reply:
I had a thought here because the issue is not the complexity of the changes, the issue is that there are just a lot of them and that's unavoidable because you can't add an assert and skip the guards or it won't build correctly. If the PR is split into two commits: One with only the gn and header asserts and one with all the gn and other buildflags added then you only have to review the first commit because the second one is driven entirely by `gn check`

Reply:
here's what I'll do.
I'll keep this PR open to track the overall task.
Then I'll split it into various smaller PRs that will be based off of each other. 
A<-B<-C...

When A passes review I won't merge to master.
When B passes review I'll merge to A.
...
Then at the end it'll be the same as this PR. And I can merge A at that point.

Reply:
Split into 6 like this:

```
c5583d77be1 (HEAD -> wallet-build-flag, origin/wallet-build-flag) Add static_assert for ENABLE_BRAVE_WALLET in wallet headers
93761fdf854 Guard wallet usage in non-wallet Brave code
2754ac22359 Guard wallet UI code in front-end TypeScript and Mojom
e191a57a46a Guard wallet code in Chromium source overrides
8607d9a6700 Add patches for wallet buildflags in Chromium BUILD files
70ac9b1abe2 Add enable_brave_wallet GN variable and build flag
```

I'll split those commits into 1 PR per commit next, and then you can review them one at a time. Each one will be based off of each-other and they will only merge into each other until all pass.